accepted/0000-workspaces-part2.md

```diff
@@ -0,0 +1,221 @@
+# npm workspaces: Running commands
+
+## Summary
+
+Introduces basic support to running **npm commands** across nested **workspaces**.
+
+## Motivation
+
+The need for running **npm commands** across nested **workspaces** was sourced from the community during the elaboration of the [original **npm workspaces** RFC](https://github.com/npm/rfcs/pull/103).
+
+## Detailed Explanation
+
+Following up from the original **npm workspaces** RFC are the following command-related changes:
+
+- A subset of the standard **npm commands** needs to be made aware of **npm workspaces**
+- It's desired to have a standard way to filter out **workspaces** in which to run specific **npm commands**
+
+## Rationale and Alternatives
+
+The ability to run **npm commands** across defined **workspaces** is essential to successfully manage a **npm workspaces** workflow. That makes the following alternatives much less desireable:
+
+- Do not implement further support to manage running commands across **workspaces**
+- Implement only _some_ of the implementation changes described in this document
+
+**Note:** Alternative ways of executing each of the described features are defined along with their implementation details below.
+
+## Implementation
+
+### 1. Run commands across all child packages
+
+Make **npm cli** subcommands **npm workspaces**-aware, so that running a command tries to run it within all **workspaces** as long as a **workspaces configuration** field is properly defined in `package.json`.
+
+Only a subset of commands are to be supported:
+
+- `fund` List funding info for all **workspaces**
+- `ls` List all packages including **workspaces**
+- `outdated` List outdated **dependencies** including **workspaces** and its **dependencies**
+- `run-script` Run arbitrary **scripts** in all **workspaces**, skip any **workspace** that does not have a targetting **script**
```

Comment:
I would highly suggest a number of different options to configure the parallelization of `run`

- `--parallel` spawn scripts across different packages in parallel processes (default to [physical cores](https://www.npmjs.com/package/physical-cpu-count))
- `--parallel=<count>` spawn `<count>` processes (as much as possible)
- `--parallel-nodes` split workload automatically across [CI Nodes](https://github.com/jamiebuilds/ci-parallel-vars)
- `--parallel-order=dependency-graph` limit parallelization by workspace cross-dependency graph, so that scripts only run on packages once all other packages have been run ([see `task-graph-runner` for example](https://github.com/jamiebuilds/task-graph-runner))
- `--parallel-order=any` Run scripts on packages in any order

Workspace-based monorepos have a huge need to optimize their different scripts and these options go a huge way to solve for different use cases.